core/arch/x86_64/include/kernel/thread.h

```diff
@@ -88,7 +88,7 @@ struct thread_core_local *thread_get_core_local(void);
  *
  * Returns true on success and false on errors.
  */
-bool thread_init_stack(uint32_t stack_id, vaddr_t sp);
+bool thread_init_stack(uint32_t stack_id, vaddr_t sp, vaddr_t abt_sp);
```

Comment:
Suggest to change the patch title to "Change interrupt stack from per core to per thread".